File: 0169-majority-element/0169-majority-element.c
int majorityElement(int A[], int size) {
    int vote=1;
    int majority=A[0];
    for(int i=1;i<size;i++){
        if(vote==0){
            vote++;
            majority=A[i];
        }else if(A[i]==majority){
            vote++;
        }else{
            vote--;
        }
    }
    return majority;
}

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int vote = 0;
        int element = 0;
        for(int val : nums){
            if(vote == 0){
                element = val;
                vote = 1;
            }else if(element == val) vote++;
            else vote--;
        }
        return element;
    }
}